src/electron/electron/handler.cljs

```diff
@@ -56,21 +57,36 @@
           new-path    (str recycle-dir "/" file-name)]
       (fs/renameSync path new-path))))
 
+(defonce Diff (google-diff.))
+(defn string-some-deleted?
+  [old new]
+  (let [result (.diff_main Diff old new)]
+    (some (fn [a] (= -1 (first a))) result)))
+
+(defn- truncate-old-versioned-files!
+  [file-path]
+  (let [dir (path/dirname file-path)
+        files (fs/readdirSync dir (clj->js {:withFileTypes true}))
+        files (map #(.-name %) files)
+        prefix (str (path/basename file-path) ".")
```

Comment:
A special case for this prefix check:

what if a user creates a page "A" and then "A/md".

Reply:
basename will return the file name with the suffix, e.g. `a.md`, `prefix` here has an ending `.` so that `a.md` will not be deleted if we want to truncate the history files for the page `A`.